src/node/node.go

```diff
@@ -59,10 +59,11 @@ type Node struct {
 func NewNode(conf *Config,
 	id uint64,
 	key *ecdsa.PrivateKey,
-	participants *peers.Peers,
 	store poset.Store,
 	trans net.Transport,
-	proxy proxy.AppProxy) *Node {
+	proxy proxy.AppProxy,
+	selectorInitFunc SelectorCreationFn,
+	selectorInitArgs SelectorCreationFnArgs) *Node {
```

Comment:
selectorInitArgs better to have type of interface{}, then selectorInitFunc would be called as
```
peerSelector := selectorInitFunc(selectorInitArgs)
```
and selectorInitArgs could be of any type, not only of explicitly defined SelectorCreationFnArgs.

In particular, RandomPeerSelector do not use GetFlagTableFn field in SelectorCreationFnArgs, so it will have other type of argument without this field.

Reply:
Thanks for the input Maxim. I will make the necessary changes

Reply:
There is a problem with your suggested change in NewNode in that NewNode needs to pass in additional selector specific arguments.
This means it cannot add the **pubKey** or **GetPeerFlagTableOfRandomUndeterminedEvent**
```
pubKey := core.HexID()

selectorInitArgs.PubKey = pubKey
selectorInitArgs.GetFlagTable = core.poset.GetPeerFlagTableOfRandomUndeterminedEvent
```

Can I use a map[string]interface{} instead of interface{} for selectorInitArgs?